src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -28,66 +28,133 @@
 
 # otio
 import opentimelineio as otio
+from xml.etree import ElementTree as ET
+from collections import OrderedDict
 
 # python
 import math
 from random import seed
 from random import random
 from collections import namedtuple
 
-COLORS = {
-    'transparent': (0, 0, 0, 0),
-    'black': (0.0, 0.0, 0.0, 255.0),
-    'white': (255.0, 255.0, 255.0, 255.0),
-    'transluscent_white': (255.0, 255.0, 255.0, 178.5),
-    'purple': (127.5, 0.0, 127.5, 255.0),
-    'light_blue': (134.895, 206.04, 235.11, 255.0),
-    'blue': (0.0, 0.0, 255.0, 255.0),
-    'dark_blue': (0.0, 0.0, 137.7, 255.0),
-    'green': (0.0, 127.5, 0.0, 255.0),
-    'dark_green': (0.0, 99.45, 0.0, 255.0),
-    'yellow': (255.0, 255.0, 0.0, 255.0),
-    'gold': (255.0, 214.2, 0.0, 255.0),
-    'orange': (255.0, 164.985, 0.0, 255.0),
-    'red': (255.0, 0.0, 0.0, 255.0),
-    'dark_red': (137.7, 0.0, 0.0, 255.0),
-    'brown': (137.7, 68.85, 25.5, 255.0),
-    'pink': (255.0, 191.25, 201.45, 255.0),
-    'gray': (127.5, 127.5, 127.5, 255.0),
-    'dark_gray': (168.3, 168.3, 168.3, 255.0),
-    'dark_gray_transluscent': (168.3, 168.3, 168.3, 200.0)
-}
+ET.sorted = lambda x: x
+try:
+    ET.Element = ET._Element_Py
+
+    def SubElement(parent, tag, attrib=OrderedDict(), **extra):
+        attrib = attrib.copy()
+        attrib.update(extra)
+        element = parent.makeelement(tag, attrib)
+        parent.append(element)
+        return element
+
+    ET.SubElement = SubElement
+except AttributeError:
+    pass  # nothing else for python2, ElementTree is pure python
+
+
+class AdapterConfig:
+    all_clips_data = []
+    trackwise_clip_count = []
+    tracks_duration = []
+    track_transition_available = []
+    max_total_duration = 0
+    global_min_time = 0
+    global_max_time = 0
+    scale_x = 1.0
+    scale_y = 1.0
+    x_origin = 0
+    image_margin = 20.0
+    arrow_margin = 10.0
+    font_size = 15
+    text_margin = 0.5 * font_size
+    clip_rect_height = 0
+    vertical_drawing_index = -1
+    arrow_label_margin = 5.0
+    image_width = 1000.0
+    image_height = 660.0
+    font_family = "Arial"
+    svg_elem = None
+
 
 random_colors_used = []
 
 
-def get_random_color():
-    return [random(), random(), random()]
+class Color:
+    def __init__(self, r=0.0, g=0.0, b=0.0, a=255.0):
+        self.value = (r, g, b, a)
```

Comment:
Because all svg colors have 0-255 range, only opacity has a 0-1 range.

Reply:
I think it may be clearer for your code if you used 0-1 range and then quantize to an integer 0-255 value in your `svg_color` method - this keeps the color class agnostic of the final quantization. This also lets you simplify a lot of code where you're either multiplying or dividing by 255.